zerver/webhooks/slack/view.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_message, create_stream_if_needed
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+from zerver.models import UserProfile, Client
+
+
+@api_key_only_webhook_view('Slack')
+@has_request_variables
+def api_slack_webhook(request, user_profile, client, user_name=REQ(), text=REQ(), channel_name=REQ(),
+                      stream=REQ(default='slack'), channels_map_to_topics=REQ(default=1)):
+    # type: (HttpRequest, UserProfile, Client, str, str, str, str, int) -> HttpResponse
+
+    sender = user_name
+    message = text
+    channel = channel_name
+    channels_map_to_topics = int(channels_map_to_topics)
+
+    if channels_map_to_topics:
+        subject = "channel: " + str(channel)
+        zulip_message = "**%s** has sent the following message:\n `%s` " % (sender, message)
+        check_send_message(user_profile, client, "stream", [stream], subject, zulip_message)
+    else:
+        stream, corrected = create_stream_if_needed(user_profile.realm, channel)
```

Comment:
Do you need these variables? I mean are they being used? 
if not please just call the function (without getting their results)